aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -236,21 +186,17 @@ static Tensor& prod_out(Tensor& result, const Tensor& self, IntList dim,
   return result;
 }
 
-static Tensor prod(const Tensor& self, IntList dim, bool keepdim, optional<ScalarType> dtype) {
+Tensor prod(const Tensor &self, optional<ScalarType> dtype) {
   Tensor result;
-  native::prod_out(result, self, dim, keepdim, dtype);
+  return at::native::prod_impl(result, self, {}, false, dtype);
   return result;
 }
 
-Tensor prod(const Tensor &self, ScalarType dtype) {
-  return at::native::prod(self, {}, false, optional<ScalarType>(dtype));
-}
+// \ALL REDUCE ################################################################
```

Comment:
these comments don't seem relevant anymore.

Reply:
There was an attempt at some point but it seems to have been broken. Let me reorder by primary op.